src/js/components/SearchInput.js

```diff
@@ -155,6 +155,12 @@ export default class SearchInput extends Component {
       if (this.props.onSelect) {
         this.props.onSelect({target: this.refs.input, suggestion: suggestion});
       }
+    } else if (this.props.onSubmit) {
```

Comment:
Thanks so much for your contribution. But I don't think we need an `onSubmit` prop to solve the issue.

If you look here https://github.com/grommet/grommet/blob/master/src/js/components/SearchInput.js#L150

I believe we have to move this line to be within the if clause. As in:

```javascript
 _onEnter (event) {
    this.setState({dropActive: false});
    if (this.state.activeSuggestionIndex >= 0) {
      event.preventDefault(); // prevent submitting forms
      let suggestion = this.props.suggestions[this.state.activeSuggestionIndex];
      this.setState({value: suggestion});
      if (this.props.onSelect) {
        this.props.onSelect({target: this.refs.input, suggestion: suggestion});
      }
    }
  }
```